electronics_lib/Passives.py

```diff
@@ -314,6 +297,230 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
       raise ValueError(f"no single capacitors in ({capacitance}) F")
 
 
+class SmtCeramicCapacitorGeneric(Capacitor, FootprintBlock, GeneratorBlock):
+  @init_in_parent
+  def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+
+    self.footprint_spec = self.Parameter(StringExpr(""))
+
+    # Default to be overridden on a per-device basis
+    self.single_cap_max = 22e-6
+    self.single_nominal_capacitance = self.Parameter(RangeExpr((0, self.single_cap_max * 1.25)))  # maximum capacitance in a single part
+
+    self.generator(self.select_capacitor_no_prod_table, self.capacitance, self.voltage, self.single_nominal_capacitance,
+                   self.part_spec, self.footprint_spec)
+
+    # Output values
+    self.selected_capacitance = self.Parameter(RangeExpr())
+    self.selected_derated_capacitance = self.Parameter(RangeExpr())
+    self.selected_voltage_rating = self.Parameter(RangeExpr())
+
+  # Chosen by a rough scan over available parts on Digikey
+  # at voltages 10v, 16v, 25v, 50v, 100v, 250v
+  # and capacitances 1.0, 2.2, 4.7
+  #
+  # For Class-1 dielectric (C0G/NP0), 20% tolerance
+  # 0402: 50v/1nF
+  # 0603: 100v/1nF, 50v/2.2nF ?
+  # 0805: 100v/2.2nF, 50v/10nF
+  # 1206: 100v/10nF
+  #
+  # For Class-2 dielectric (X**), 20% tolerance
+  # 0402:                   50v /                0.1uF,     25v / 0.1uF,                      10v / 2.2uF
+  # 0603:                   50v /                0.1uF,     25v /   1uF,     16v / 2.2uF,     10v /  10uF
+  # 0805: 100v / 0.1uF,     50v / 0.1uF (maybe 0.22uF),     25v /  10uF
+  # 1206: 100v / 0.1uF,     50v /                4.7uF,     25v /  10uF,                      10v /  22uF
+  # 1210: 100v / 4.7uF,     50v /                 10uF,                      16v /  22uF,     10v /  47uF
+  # 1812: 100v / 2.2uF,     50v /                  1uF,     25v /  10uF (though small sample size)
+
+  # derating coefficients in terms of %capacitance / V over 3.6
+  # 'Capacitor_SMD:C_0603_1608Metric'  # not supported, should not generate below 1uF
+
+  class SmtCeramicCapacitorGenericPackageSpecs(NamedTuple):
+    name: str
+    max: float
+    derate: float
+    vc_pairs: dict
+
+  PACKAGE_SPECS = {
+    402: SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_0402_1105Metric',
+      max=1e-7,
+      derate=0,
+      vc_pairs={             50:   1e-7, 25: 1e-7,             10: 2.2e-6},
+    ),
+    603: SmtCeramicCapacitorGenericPackageSpecs(
+      name='Capacitor_SMD:C_0603_1608Metric',
+      max=1.1e-6,
+      derate=0,
```

Comment:
I think derating should be all-or-nothing: either it's applied and defined (even if kinda guessy) for all footprints, or none. If you find curves for 0603 / 0402, I'd recommend applying them; otherwise, consider removing it.

If you zero out the derating here, it may be worth keeping the code in case we use it in the future. However, would need to figure out how to test it.